From 0276d4ea24b456f38dd341f424ccc9029cf841cb Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Thu, 7 Jul 2022 15:44:20 +0200
Subject: [PATCH 0094/1141] arm64: dts: st: add scmi performance support on
 stm32mp25

Update the device-tree to add support for scmi performance
on cpu0 and cpu1. This leads to enable cpu-freq on the CPUs.

Signed-off-by: Pascal Paillet <p.paillet@st.com>
Change-Id: I853c03c86900c5959e93f42534a5b8bd72cf6ea0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/304313
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 7 +++++++
 arch/arm64/boot/dts/st/stm32mp253.dtsi | 2 ++
 2 files changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index fd008774dc96..3bd96a86aef7 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -22,6 +22,8 @@ cpu0: cpu@0 {
 			device_type = "cpu";
 			reg = <0>;
 			enable-method = "psci";
+			clocks = <&scmi_perf 0>;
+			clock-names = "cpu";
 		};
 	};
 
@@ -50,6 +52,11 @@ scmi: scmi {
 			#size-cells = <0>;
 			linaro,optee-channel-id = <0>;
 
+			scmi_perf: protocol@13 {
+				reg = <0x13>;
+				#clock-cells = <1>;
+			};
+
 			scmi_clk: protocol@14 {
 				reg = <0x14>;
 				#clock-cells = <1>;
diff --git a/arch/arm64/boot/dts/st/stm32mp253.dtsi b/arch/arm64/boot/dts/st/stm32mp253.dtsi
index 89dd2a54f90d..3cff3c4c1f5b 100644
--- a/arch/arm64/boot/dts/st/stm32mp253.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp253.dtsi
@@ -12,6 +12,8 @@ cpu1: cpu@1 {
 			device_type = "cpu";
 			reg = <1>;
 			enable-method = "psci";
+			clocks = <&scmi_perf 0>;
+			clock-names = "cpu";
 		};
 	};
 
-- 
2.39.2

